### PR TITLE
12554-Move-creation-of-ObjectMapper-to-JacksonMapperUtilities

### LINKED ISSUE #12554: Move creation of ObjectMapper to JacksonMapperUtilities or companion objects
### User Story
Instantiating an `ObjectMapper` is a pretty heavy operation that should only occur once per configuration. We should hold onto the few ObjectMappers we need rather than creating new ones on the fly all over the codebase.

ObjectMappers are completely thread-safe and do not need to instantiated per read/write.

### Risks/Impacts/Considerations
Using the wrong mapper in the initial spot which could cause problems serializing/deserializing.

### Dev Notes
Look for the following calls:
```kotlin
ObjectMapper()
jacksonObjectMapper()
jacksonMapperBuilder()
```

If you are using a plain json mapper use the existing  one in `JacksonMapperUtilities`.

We should add a default configuration for a YAML `ObjectMapper` in `JacksonMapperUtilities` and use it wherever default YAML parsing is needed.

Mappers created in companion objects are generally the best option if the code you are working with needs a custom configuration. 

### Acceptance Criteria
- New default YAML `ObjectMapper`
- Fewer instantiations of `ObjectMapper` and depending on the existing ones in `JacksonMapperUtilities`.
- Custom `ObjectMapper` in companion object or investigate if the custom one is really necessary

### PR DESCRIPTION
**Pull request by JFU-NAVA-PBC**
This PR:

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Applies singleton pattern to the usage of jackson object mapper.

Test Steps:
1. Checkout PR to local repo
2. Run and pass all tests

Profiled (gradle profiler) the end2end_up PR vs master, make sure no negative impact on latencies:

PR elapsed time avg:

image

Master elapsed time avg:

image

 
## Changes
- Added singleton object mappers of flavors into JacksonMapperUtilities
- Refactored code base - main code and tests where direct ObjectMapper / jsonObjectMapper constructors used, replace with the singleton usage in JacksonMapperUtilities.

### Testing
- [X] Tested locally?
- [X] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #12554

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #12800

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/kotlin/azure/Event.kt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package gov.cdc.prime.router.azure
 
-import com.fasterxml.jackson.databind.ObjectMapper
+// import com.fasterxml.jackson.databind.ObjectMapper
```
Comment (jalbinson) [inline prime-router/src/main/kotlin/azure/Event.kt:3]: You can delete this commented out import if no longer needed.

File: prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt [hunk 2]
```diff
@@ -497,7 +497,6 @@ hnm8COa8Kr+bnTqzScpQuOfujHcFEtfcYUGfSS6HusxidwXx+lYi1A==
 
     @Test
     fun `test flexion rest transport`() {
-        val mockHttpClient = mockk<HttpClient>()
```
Comment (jalbinson) [inline prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt:500]: Was this just unused? Seems unrelated to the PR.

**Review by jalbinson — APPROVED** [review]
Looks good. Just need a few tweaks and it's ready to be merged.

File: prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt (hunk 2 above)
Comment (JFU-NAVA-PBC) [reply prime-router/src/testIntegration/kotlin/transport/RESTTransportIntegrationTests.kt:500]: think so, the linter complained about it

File: prime-router/src/main/kotlin/common/JacksonMapperUtilities.kt [hunk 3]
```diff
@@ -50,6 +51,16 @@ object JacksonMapperUtilities {
         enable(SerializationFeature.INDENT_OUTPUT) // Pretty print JSON output
     }.run { setTimeZone(TimeZone.getTimeZone(Environment.rsTimeZone)) }
 
+    /**
+     * Mapper using library defaults (no kotlin, java modules).
+     */
+    val objectMapper: ObjectMapper = ObjectMapper()
+
+    /**
+     * jacksonObjectMapper using library defaults (no kotlin, java modules).
```
[the remark this answers is not in the document]
Comment (JFU-NAVA-PBC) [reply prime-router/src/main/kotlin/common/JacksonMapperUtilities.kt:60]: good catch

**Review by arnejduranovic — APPROVED** [review]
Changes look good, object mapper created in static class and reused throughout.

**Pull request merged**